### DISCUSSION TITLE
Input value isn't editable after submitting

### DISCUSSION THREAD

**Discussion by JenniferFalkenstein** [react-hook-form/react-hook-form#5577] [Q&A]
I got four input fields, which are assigned with `register`.
When I enter some input into those fields and click the "continue"-Button, the code writes those input-values into an object.
But when I click back and want to edit the entries I just did, you can't edit it anymore. It just stays with the old value.
Apparently it's the `register`, because I tried it without `register` but with `onChange`, which works fine.

Here's a part of my code:
`const [contactData, setProjectData] = React.useState({ email: undefined  });`
`const { register, handleSubmit, formState: { errors } } = useForm();`

` <div>`
`<input {...register('name', { required: true, pattern: /[A-Za-z]/ })} placeholder='Your name' value={contactData.name} />`
`{errors.name && (<span role='alert'>Name ist erforderlich!</span>)}`
`<div>`
`<button onClick={handleSubmit(onSubmit)>Continue</button>`

and then somewhere the function:
`const onSubmit = (newData) => {`
`    updateData(Object.assign(contactData, newData));`
`    showNextPage();`
`  };`

**Comment by bluebill1049** [comment]
can you provide a CSB?

**Reply by JenniferFalkenstein** [reply]
I'm sorry, but what's a CSB? 😓

**Reply by bluebill1049** [reply]
codesandbox.

**Reply by JenniferFalkenstein** [reply]
Oh right, makes sense.
Although, there's no need for that anymore, since I tried to set it up on codesandbox and found the solution to this issue.
It wasn't `react-hook-form`'s fault, I just needed to change `value` into `defaultValue`

Sorry for that and thanks alot for trying to help 👏

**Reply by bluebill1049** [reply]
no worries :) i am glad you found out the issue @JenniferFalkenstein
